api4/post.go

```diff
@@ -581,6 +581,19 @@ func getPostThread(c *Context, w http.ResponseWriter, r *http.Request) {
 		c.Err = err
 		return
 	}
+	if list.FirstInaccessiblePostTime != 0 {
+		clientPostList := c.App.PreparePostListForClient(c.AppContext, list)
+		clientPostList, err = c.App.SanitizePostListMetadataForUser(c.AppContext, clientPostList, c.AppContext.Session().UserId)
+		if err != nil {
+			c.Err = err
+			return
+		}
+
+		if err := clientPostList.EncodeJSON(w); err != nil {
+			mlog.Warn("Error while writing response", mlog.Err(err))
```

Comment:
Minor side note: please avoid using mlog.Warn in `api4` layer. Instead use `c.Logger.(Warn|Info|Error)`. We are trying to move away from the global logger and use the context based logger.

cc @noxer